Client/Frontend/Library/BookmarksPanel.swift

```diff
@@ -255,16 +272,22 @@ class BookmarksPanel: SiteTableViewController, LibraryPanel {
         if bookmarkNodes[safe: indexPath.row] is BookmarkSeparator {
             return BookmarksPanelUX.SeparatorRowHeight
         }
-
         return super.tableView(tableView, heightForRowAt: indexPath)
     }
 
     override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
-        return nil
+        guard section > 0, recentBookmarks.count > 0 else {
+            return nil
+        }
+        return super.tableView(tableView, viewForHeaderInSection: section)
+    }
+
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return section == 0 ? nil : Strings.RecentlyBookmarkedTitle
```

Comment:
I would double-check with Bryan to see if he wants a header for the root folders too (i.e. "Locations").